### ISSUE TITLE
[TEST] model_ops_v2 factory ops with a top-level `device` kwarg run on CPU, not on the device

### ISSUE THREAD

**Issue by yoheiueda** [torch-spyre/torch-spyre#3894]
### What happened?

Ops declared in the `model_ops_tests` configs with `device` as a **top-level kwarg** of `sample_inputs_func` never execute on the Spyre device. They run entirely on CPU, and the harness's own device assertion is disarmed in the same motion, so the test reports success rather than flagging that the device was never touched.

The reported case is `torch.arange` in `tests/configs/model_ops_tests/gpt-oss-20b_spyre.yaml`, declared as:

```yaml
- name: torch.arange
  sample_inputs_func:
    args:
      - value: 11
    kwargs:
      device: cuda:0
```

Under the reported command this executes `torch.arange(11, device='cpu')` on both the reference and the "device" side.

Three separate pieces interact to produce this, and the third is what makes it silent.

**1. The sample is built once, for CPU, and that collapses the device kwarg.** `tests/models/test_model_ops_v2.py:411-415` builds the sample with a hardcoded CPU target:

```python
cpu_sample: SampleInput = ops_item.build_sample_input(
    seed=seed,
    test_device="cpu",
    SampleInput=SampleInput,
)
```

`InputsEdits.resolved_kwargs` then applies its documented device-replacement rule (`tests/oot_framework/oot_test_config_models.py:1096-1098`):

```python
# 2. device replacement
if k == "device" and test_device is not None and "cuda" in v:
    out[k] = test_device
    continue
```

Because `test_device` is `"cpu"` at this call, the YAML's `cuda:0` resolves to `cpu`. The intended device substitution is never requested:

```
resolved with test_device=cpu   : {'device': 'cpu'}
resolved with test_device=spyre : {'device': device(type='spyre')}
```

**2. The device sample is derived from the CPU one and cannot repair it.** `test_sample` is built by mapping `_to_target_device` over the CPU sample's kwarg *values* (`test_model_ops_v2.py:484`):

```python
kwargs={k: _to_target_device(v) for k, v in cpu_sample.kwargs.items()},
```

`_to_target_device` relocates **only tensors**; its final branch is a bare `return x` for everything else. A device string or `torch.device` object is not a tensor, so it falls straight through unchanged:

```
'cpu'                -> 'cpu'
device(type='cpu')   -> device(type='cpu')
```

The comment above that line notes kwargs are "plain values only (dim, dtype, keepdim, etc.)", but `device` is precisely a plain value that *does* need translating — it is the one plain kwarg whose value selects where the op runs.

**3. The guard that should catch a CPU output instead blesses it.** `_is_cpu_output` (`test_model_ops_v2.py:296-303`) keys off the already-corrupted resolved value:

```python
device_str = str(test_sample.kwargs.get("device", ""))
return device_str == "cpu" or (device_str == "" and op_name in _FACTORY_OPS)
```

Since step 1 set `device_str` to `"cpu"`, this returns `True`, `expected_device` becomes `cpu`, and the `_confirm_device` assertion at `:509` passes. Step 1 breaks the input and step 3 lowers the bar to match, so nothing reports a problem.

### What did you expect to happen?

`device: cuda:0` in a `*_spyre.yaml` config should resolve to the Spyre device for the device-side run, so that `torch.ara
nge(11, device=...)` actually allocates and computes on Spyre and is compared against a CPU reference.

Correspondingly, `_is_cpu_output` should only be true when the config **author** wrote `cpu`, so it cannot be satisfied by
 a device kwarg that was mis-resolved upstream.

### How can we reproduce it?

`device: cuda:0` in a `*_spyre.yaml` config should resolve to the Spyre device for the device-side run, so that `torch.ara
nge(11, device=...)` actually allocates and computes on Spyre and is compared against a CPU reference.

Correspondingly, `_is_cpu_output` should only be true when the config **author** wrote `cpu`, so it cannot be satisfied by
 a device kwarg that was mis-resolved upstream.

## How can we reproduce it?

The reported command (also reproduces without `--junit-xml`):

```bash
env bash tests/run_test.sh tests/configs/model_ops_tests/gpt-oss-20b_spyre.yaml -k torch_arange -v
```

Observed result — the op runs on CPU, and the run is reported as passing (`XPASS`, see note under "Anything else"):

```
test_model_ops_v2.py::TestSpyreModelOpsPRIVATEUSE1::test_model_ops_db_torch_arange__45_spyre_float16 XPASS
===================== 234 deselected, 1 xpassed in 26.73s ======================
[torch_oot_device_tests_run] Done. Overall exit code: 0
```

Step 1 in isolation — the device kwarg collapsing to `cpu`:

```bash
python - <<'EOF'
import sys; sys.path.insert(0, 'tests')
import torch
from oot_framework.oot_test_config_models import InputsEdits

ie = InputsEdits(args=[{'value': 11}], kwargs={'device': 'cuda:0'})
print('resolved with test_device=cpu   :', ie.resolved_kwargs(test_device='cpu'))
print('resolved with test_device=spyre :', ie.resolved_kwargs(test_device=torch.device('spyre')))
EOF
```

Output:

```
resolved with test_device=cpu   : {'device': 'cpu'}
resolved with test_device=spyre : {'device': device(type='spyre')}
```

Step 2 in isolation — `_to_target_device`'s non-tensor fallthrough leaves the device value untouched, so the CPU-resolved
kwarg survives into the device sample:

```bash
python - <<'EOF'
import torch

def to_target(x, test_device=torch.device('spyre')):
    if torch.is_tensor(x):
        return x.to(test_device)
    if isinstance(x, (list, tuple)):
        return type(x)(to_target(i) for i in x)
    if isinstance(x, dict):
        return {k: to_target(v) for k, v in x.items()}
    return x   # device string / torch.device hits this

for v in ['cpu', torch.device('cpu')]:
    print(f'{v!r:22} -> {to_target(v)!r}')
EOF
```

Output:

```
'cpu'                  -> 'cpu'
device(type='cpu')     -> device(type='cpu')
```

### Any environmental details we need to know?

- torch: `2.13.0+cpu`
- Python: `3.12.13`
- torch-spyre commit: `be1328a8` (branch `main`)
- pytest: `9.1.1`
- platform: Linux x86_64

### Anything else we need to know?

**Scope — not specific to `arange`.** Any op whose `device` arrives as a top-level kwarg is affected. `device` nested *inside* a tensor spec is handled correctly, because those tensors do go through `_to_target_device`; only the top-level plain kwarg is missed. Counting entries across `tests/configs/model_ops_tests/*_spyre.yaml`:

| config | device-kwarg ops | of which factory ops |
|---|---:|---:|
| `Meta-Llama-3.1-8B-Instruct_spyre.yaml` | 3 | 2 |
| `Ministral-3-14B-Instruct-2512_spyre.yaml` | 3 | 2 |
| `Mistral-Small-3.2-24B-Instruct-2506_spyre.yaml` | 5 | 3 |
| `Qwen2.5-7B-Instruct_spyre.yaml` | 2 | 2 |
| `gemma-4-12B-it_spyre.yaml` | 2 | 2 |
| `gemma-4-26B-A4B-it_spyre.yaml` | 2 | 2 |
| `gpt-oss-20b_spyre.yaml` | 3 | 3 |
| `granite-3.3-8b-instruct_spyre.yaml` | 3 | 2 |
| `granite-4.1-8b_spyre.yaml` | 3 | 2 |
| **total** | **26** | **20** |

The factory ops involved are the `_FACTORY_OPS` set in `test_model_ops_v2.py:199-210` (`torch.arange`, `torch.zeros`, `torch.ones`, `torch.full`, `torch.randn`, `torch.randint`, `torch.empty`, ...).

**Current CI impact is limited, but the coverage is illusory.** These cases are xfail-marked today, so they surface as `XPASS` rather than `PASS` and are not yet asserting a false green. The concern is that they provide no device coverage at all while appearing to, and if the xfail marking is removed a genuine device-side regression in these ops would pass unnoticed.

**Suggested fix direction.** Resolve kwargs per target instead of deriving the device sample from the CPU sample — i.e. build the sample a second time with the real `test_device` so the existing rule 2 substitutes the Spyre device. Then change `_is_cpu_output` to key off the **original YAML** value (did the author write `cpu`?) rather than the resolved one, so it can no longer be satisfied by a mis-resolved device. Fixing only step 1 without step 3 would leave the weakened assertion in place; fixing only step 3 would turn these into hard failures without making them actually run on the device.

### Relevant log output

```Shell
$ env bash tests/run_test.sh tests/configs/model_ops_tests/gpt-oss-20b_spyre.yaml -k torch_arange -v

[torch_oot_device_tests_run] Running serial test
============================= test session starts ==============================
platform linux -- Python 3.12.13, pytest-9.1.1, pluggy-1.6.0
rootdir: /home/yohei/tasks/worktrees/main/torch-spyre
configfile: pyproject.toml
plugins: anyio-4.14.2, xdist-3.8.0
collecting ... collected 235 items / 234 deselected / 1 selected

test_model_ops_v2.py::TestSpyreModelOpsPRIVATEUSE1::test_model_ops_db_torch_arange__45_spyre_float16 XPASS [TAGS = model__gpt-oss-20b_spyre dtype__float16 op__torch_arange platform__x86_64 testtype__trunk constant torch.arange.1_spyre] [100%]
  [INPUT SHAPES]
  arg[0]: value=11

===================== 234 deselected, 1 xpassed in 26.73s ======================

[torch_oot_device_tests_run] Per-file result summary:
  PASS      test_model_ops_v2.py                                  1 xpassed in 26.73s
[torch_oot_device_tests_run] Done. Overall exit code: 0
```
